python/lsst/ip/isr/isrFunctions.py

```diff
@@ -592,6 +594,230 @@ def brighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain, ga
+        change in the direction of the effect)
+
+    Returns
+    -------
+    diff : `float`
+        Final difference between iterations achieved in correction.
+    iteration : `int`
+        Number of iterations used to calculate correction.
+
+    Notes
+    -----
+    Modified version of brighterFatterCorrection()
+    This correction takes a kernel that has been derived from flat
+    field images to redistribute the charge.  The gradient of the
+    kernel is the deflection field due to the accumulated charge.
+    Given the original image I(x) and the kernel K(x) we can compute
+    the corrected image Ic(x) using the following equation:
+    Ic(x) = I(x) + 0.5*d/dx(I(x)*d/dx(int( dy*K(x-y)*I(y))))
+    Improved algorithm at this step applies the divergence theorem to
+    obtain a pixelised correction.
+    Because we use the measured counts instead of the incident counts
+    we apply the correction iteratively to reconstruct the original
+    counts and the correction.  We stop iterating when the summed
+    difference between the current corrected image and the one from
+    the previous iteration is below the threshold.  We do not require
+    convergence because the number of iterations is too large a
+    computational cost.  How we define the threshold still needs to be
+    evaluated, the current default was shown to work reasonably well
+    on a small set of images.
+    Edges are handled in the convolution by padding.  This is still not
+    a physical model for the edge, but avoids discontinuity in the correction.
+
+    Author of modified version: Lance.Miller@physics.ox.ac.uk
+    (see DM-33555).
+    """
+
+    image = exposure.getMaskedImage().getImage()
+
+    # The image needs to be units of electrons/holes
+    with gainContext(exposure, image, applyGain, gains):
+
+        # get kernel and its shape
+        kLy, kLx = kernel.shape
+        kernelImage = afwImage.ImageD(kLx, kLy)
+        kernelImage.getArray()[:, :] = kernel
+        tempImage = image.clone()
+
+        nanIndex = numpy.isnan(tempImage.getArray())
+        tempImage.getArray()[nanIndex] = 0.
+
+        outImage = afwImage.ImageF(image.getDimensions())
+        corr = numpy.zeros_like(image.getArray())
+        prevImage = numpy.zeros_like(image.getArray())
+        convCntrl = afwMath.ConvolutionControl(False, True, 1)
+        fixedKernel = afwMath.FixedKernel(kernelImage)
+
+        # set the padding amount
+        # ensure we pad by an even amount larger than the kernel
+        kLy = 2 * ((1+kLy)//2)
+        kLx = 2 * ((1+kLx)//2)
+
+        # The deflection potential only depends on the gradient of
+        # the convolution, so we can subtract the mean, which then
+        # allows us to pad the image with zeros and avoid wrap-around effects
+        # (although still not handling the image edges with a physical model)
+        # This wouldn't be great if there were a strong image gradient.
+        imydim, imxdim = tempImage.array.shape
```

Comment:
Can these be renamed to `imYdim` and `imXdim`?  I'm fine with any other name, but without some separation, I read the first one as i-my-dim, which confused me.